src/message/headers/TopicMessageHeader.js

```diff
@@ -38,15 +38,30 @@ export default class TopicMessageHeader extends React.PureComponent {
     doNarrow(topicNarrow(stream, topic), itemId);
   }
 
+  topicText() {
+    const { topic, isTopicNarrow } = this.props;
+    if (isTopicNarrow) {
```

Comment:
If topic is long enough, then it may not be able to fit in a single line. Web supports topic with 60 char  . So for such cases, i thought to have topic multiline, only in topic narrow. Home view (below image) and stream narrow looks pretty good with single line topic and `ellipsizeMode="tail"`.
![070d97d8-2421-4e22-ab03-726957a9ba13](https://cloud.githubusercontent.com/assets/18511177/25111113/ab91b0ae-2406-11e7-8a2c-f67026bb5244.jpg)


But if topic is quite long and does not fit in single line (whole line) then there should be some place where user can get full topic name (I found topic  narrow  is the best place because it is the view where full focus is on a single topic.).

![d2857612-ec0d-48df-aa27-702369bf44ca](https://cloud.githubusercontent.com/assets/18511177/25111135/c7d6b6ec-2406-11e7-9612-852a83843b8b.jpg)

If we go with `numberOfLines={1} ellipsizeMode="tail"` even in topic narrow then entire topic (long) might not be visible, it is more frequent on devices with small screen. 
![8299201e-48ec-4e76-be61-612490016492](https://cloud.githubusercontent.com/assets/18511177/25111299/c42516d2-2407-11e7-8eaf-54c642e2119c.jpg)